pkg/helm/release/manager.go

```diff
@@ -241,43 +242,48 @@ func reconcileRelease(ctx context.Context, kubeClient kube.Interface, expectedMa
 	}
 	return expectedInfos.Visit(func(expected *resource.Info, err error) error {
 		if err != nil {
-			return err
+			return fmt.Errorf("visit error: %w", err)
 		}
 
 		expectedClient := resource.NewClientWithOptions(expected.Client, func(r *rest.Request) {
 			*r = *r.Context(ctx)
 		})
 		helper := resource.NewHelper(expectedClient, expected.Mapping)
 
-		existing, err := helper.Get(expected.Namespace, expected.Name, false)
+		existing, err := helper.Get(expected.Namespace, expected.Name, expected.Export)
 		if apierrors.IsNotFound(err) {
 			if _, err := helper.Create(expected.Namespace, true, expected.Object,
 				&metav1.CreateOptions{}); err != nil {
 				return fmt.Errorf("create error: %s", err)
 			}
 			return nil
 		} else if err != nil {
-			return err
+			return fmt.Errorf("could not get object: %w", err)
 		}
 
-		patch, err := generatePatch(existing, expected.Object)
+		expectedObj, err := asVersioned(expected)
```

Comment:
I tried to run your branch in our cluster and noticed an error:

`could not get versioned object: failed to convert to versioned object: no kind \"CustomResource\" is registered for version \"example.com/v1alpha1\" in scheme \"pkg/runtime/scheme.go:101\""`

It seems that strategic merge patches are not supported on CRs :(
https://github.com/kubernetes/kubernetes/issues/58414
https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#advanced-features-and-flexibility

We are currently working on a fix to fallback to the previous json patch method
@hensur

Reply:
Good catch! 

> We are currently working on a fix to fallback to the previous json patch method

Sounds good! Feel free to post another PR that supercedes this one if you figure something out.

/hold